src/microbe_stage/systems/ProcessSystem.cs

```diff
@@ -192,6 +203,23 @@ public sealed class ProcessSystem : AEntitySetSystem<float>
                     processATPProduction += amount;
 
                     result.AddProduction(organelle.Definition.InternalName, amount);
+
+                    var isInEnvironment = true;
+                    foreach (var input in processData.WritableInputs)
```

Comment:
The performance of this slightly concerns me. I think an improvement would be to first check / precalculate if the process has environmental inputs, and only then run this piece of code.

Reply:
This is probably a case of bad variable naming on my part, but this check is instead checking that the inputs for the process are present in that Patch at all, not if they are environmental. isInPatch would likely be a better name.

Reply:
But wouldn't it then make more sense to always check that the process can run even theoretically in the patch? Because shouldn't energy balance contribution be 0 from processes that have no chance of running at all?